### PR TITLE
Add support for Spring Cloud LoadBalancer

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**

### CODE REVIEW

File: start-site/src/main/resources/application.yml [hunk 1]
```diff
@@ -1028,6 +1028,16 @@ initializr:
           description: Declarative REST Client. OpenFeign creates a dynamic implementation of an interface decorated with JAX-RS or Spring MVC annotations.
           groupId: org.springframework.cloud
           artifactId: spring-cloud-starter-openfeign
+        - name: Spring Cloud LoadBalancer
+          id: cloud-loadbalancer
+          compatibilityRange: "[2.2.0.M4,2.2.x.BUILD-SNAPSHOT)"
```
Comment (snicoll) [inline start-site/src/main/resources/application.yml]: Is there a reason why the entry should be disabled with snapshots? (Compared to the rest of Spring Cloud I mean).

File: start-site/src/main/resources/application.yml [hunk 2]
```diff
@@ -1028,6 +1028,16 @@ initializr:
...
+          description: Client-side load-balancing with Spring Cloud LoadBalancer.
+          groupId: org.springframework.cloud
+          artifactId: spring-cloud-starter-loadbalancer
+          links:
+            - rel: guide
```
Comment (snicoll) [inline start-site/src/main/resources/application.yml]: Can we please have a link to the documentation as well?

File: start-site/src/main/resources/application.yml (hunk 1 above)
Comment (OlgaMaciaszek) [reply start-site/src/main/resources/application.yml]: No, for 2.2.x snapshots it should be enabled - if that's the result of it, then it's a bug; should this look differently to include 2.2.x snapshots?

Comment (snicoll) [reply start-site/src/main/resources/application.yml]: That code says the entry should be enabled beteen `2.2.0.M4` (inclusive) to the latest snapshot of the 2.2 line (excluded). We don't use such range but rather the version one should use at the bare minimum.

File: start-site/src/main/resources/application.yml (hunk 2 above)
Comment (OlgaMaciaszek) [reply start-site/src/main/resources/application.yml]: It's not a separate project, so documentation for it is included in the SC Commons docs - there are different bits related to using it in different scenarios:
- https://cloud.spring.io/spring-cloud-static/spring-cloud-commons/2.2.0.M3/reference/html/#spring-resttemplate-as-a-load-balancer-client
- https://cloud.spring.io/spring-cloud-static/spring-cloud-commons/2.2.0.M3/reference/html/#spring-resttemplate-as-a-load-balancer-client
- https://cloud.spring.io/spring-cloud-static/spring-cloud-commons/2.2.0.M3/reference/html/#passing-your-own-load-balancer-client-configuration
Should I just add all of them?

Comment (snicoll) [reply start-site/src/main/resources/application.yml]: Can we rephrase the doc a bit to have a dedicated section for it please? If there is a dedicated starter, that's what I would expect.

File: start-site/src/main/resources/application.yml (hunk 1 above)
Comment (OlgaMaciaszek) [reply start-site/src/main/resources/application.yml]: ah, ok. thanks

File: start-site/src/main/resources/application.yml (hunk 2 above)
Comment (OlgaMaciaszek) [reply start-site/src/main/resources/application.yml]: Have merged the changes: https://github.com/spring-cloud/spring-cloud-commons/pull/624 Once we do a new release (should be this week) and this gets published, will add it here.

**Pull request closed without merging**